src/gui_common/charts/line/LineChartData.cs

```diff
@@ -1,73 +1,18 @@
-﻿using System.Collections.Generic;
-using Godot;
-
-/// <summary>
-///   Dataset to be visualized on the line chart. Contains series of data points.
+﻿/// <summary>
+///   Dataset to be visualized on <see cref="LineChart"/>. Contains series of data points.
 /// </summary>
-public class LineChartData
+public class LineChartData : ChartDataSet
 {
-    private List<DataPoint> dataPoints = new List<DataPoint>();
-    private Color dataColour;
-    private bool draw = true;
-
-    public IReadOnlyCollection<DataPoint> DataPoints => dataPoints;
-
-    /// <summary>
-    ///   The icon on the chart legend
-    /// </summary>
-    public Texture IconTexture { get; set; }
-
     public float LineWidth { get; set; } = 1.13f;
 
-    /// <summary>
-    ///   Used to differentiate the data set's visual by color
-    /// </summary>
-    public Color DataColour
+    public override object Clone()
     {
-        get => dataColour;
-        set
-        {
-            dataColour = value;
+        var result = new LineChartData();
 
-            foreach (var point in DataPoints)
-            {
-                point.MarkerColour = dataColour;
-                point.Update();
-            }
-        }
-    }
-
-    /// <summary>
-    ///   If this is true, visuals will be drawn (e.g lines, markers)
-    /// </summary>
-    public bool Draw
-    {
-        get => draw;
-        set
-        {
-            draw = value;
+        ClonePropertiesTo(result);
```

Comment:
I see ClonePropertiesTo is used here, but then why is the Clone in ChartDataSet implemented using duplicate functionality (the property copying)?